plutus-benchmark/bench-validation/Main.hs

```diff
@@ -0,0 +1,142 @@
+
+-- Generate an HTML report.  If run via stack/cabal this will be written to the
+-- `plutus-benchmark` directory.
+config :: Config
+config = defaultConfig
+  { reportFile = Just "report.html"
+  }
+
+fromDeBruijn ::  UPLC.Program UPLC.DeBruijn PLC.DefaultUni a ->  IO (Program a)
+fromDeBruijn prog = do
+    let namedProgram = UPLC.programMapNames (\(UPLC.DeBruijn ix) -> UPLC.NamedDeBruijn "v" ix) prog
+    case PLC.runQuote $ runExceptT $ UPLC.unDeBruijnProgram namedProgram of
+      Left e  -> error $ show e
+      Right p -> return p
+
+loadPlc :: Serialise a => FilePath -> IO (Program a)
+loadPlc file = do
+  BSL.readFile file <&> deserialiseOrFail >>= mapM fromDeBruijn >>= \case
+               Left (DeserialiseFailure offset msg) ->
+                   error ("Deserialisation failure at offset " ++ Prelude.show offset ++ ": " ++ msg)
+               Right r -> return r
+
+benchCek :: Term () -> Benchmarkable
+benchCek program = nf (UPLC.unsafeEvaluateCek getStringBuiltinMeanings defaultCostModel) program
+
+cborSuffix :: String
+cborSuffix = ".cbor"
+
+dataDir :: String
+dataDir = "bench-validation" </> "data"
+
+{- Construct an applied validator.  We assume that relevant validators, datum
+   scripts, redeemers and contexts are stored in CBOR format under `<progName>`
+   in the `data` directory.  These should have names like "Redeemer01.cbor",
+   "Context03.cbor", and so on. This function returnes a Criterion environment
+   to be fed to the relevant benchmark, to keep the IO overhead out of the
+   benchmark itself. -}
+getAppliedScript :: String -> Int -> Int -> Int -> Int -> IO (Term ())
+getAppliedScript progName validatorNumber datumNumber redeemerNumber contextNumber = do
+  let dataPath = dataDir </> progName
+      loadScript base suffix = do
+          let file = dataPath </> (base ++ printf "%02d" suffix ++ cborSuffix)
+          loadPlc file
+  validator <- loadScript "Validator" validatorNumber
+  datum     <- loadScript "Datum"     datumNumber
+  redeemer  <- loadScript "Redeemer"  redeemerNumber
+  context   <- loadScript "Context"   contextNumber
+  let appliedValidator = validator `UPLC.applyProgram` datum `UPLC.applyProgram` redeemer `UPLC.applyProgram` context
+  pure $ void . UPLC.toTerm $ appliedValidator
+
+
+{- Create a benchmark with a name like "crowdfunding/5" by applying validator
+   number v to datum d, redeemer r, and context c in the directory
+   data/<dirname>.  The 'id' argument is just to give the indvidual benchmarks
+   more readable names. -}
+mkBM :: String -> (Int, (Int, Int, Int, Int)) -> Benchmark
+mkBM dirname (id, (v,d,r,c)) =
+    env (getAppliedScript dirname v d r c) $ \ ~ script -> bench (show id) $ benchCek script
+
+-- Make a `bgroup` collecting together a number of benchmarks for the same contract
+mkBgroup :: String -> [(Int, (Int, Int, Int, Int))] -> Benchmark
+mkBgroup dirname bms = bgroup dirname (map (mkBM dirname) bms)
+
+main :: IO ()
+main = defaultMainWith config
```

Comment:
What I would like to know is how did you generate the CBOR files?

Reply:
I updated the plc command so that it could produce suitable CBOR (see the end of my initial comment on this PR) and used that.